src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -960,4 +1075,52 @@ public void visitToken(DetailAST ast) {
             throw new IndexOutOfBoundsException("test");
         }
     }
+
+    private static class DummyCheck extends AbstractFileSetCheck {
+
+        private final List<String> methodCalls = new ArrayList<>();
+
+        private boolean isInitCalled;
+
+        @Override
+        public void init() {
+            super.init();
+            isInitCalled = true;
+        }
+
+        @Override
+        public void beginProcessing(String charset) {
+            methodCalls.add("beginProcessing");
+            super.beginProcessing(charset);
+        }
+
+        @Override
+        public void finishProcessing() {
+            methodCalls.add("finishProcessing");
+            super.finishProcessing();
+        }
+
+        @Override
+        protected void processFiltered(File file, List<String> lines) throws CheckstyleException {
+            methodCalls.add("processFiltered");
+        }
+
+        @Override
+        public void destroy() {
+            methodCalls.add("destroy");
+            super.destroy();
+        }
+
+        public List<String> getMethodCalls() {
+            return Collections.unmodifiableList(methodCalls);
+        }
+
+        public boolean getIsInitCalled() {
+            return isInitCalled;
+        }
+
+        public boolean messageDispatcherIsEqualsTo(Object object) {
+            return getMessageDispatcher().equals(object);
```

Comment:
just make it to return `return getMessageDispatcher();` code will be clear. No incapsulation is required for Dummy classes